### PR TITLE
Datumises Wrestlers

### PR DESCRIPTION
**Pull request by Mister-Moriarty**
[Gamemodes] [Internal] [Code Quality]

## About the PR:
Wrestlers have been fully migrated onto the new datum system, including roundstart, random event, wrestlerdoodle, VR, and admin-spawned wrestlers.
Removes `is_inherent` checks from wrestler related equipment, as wrestler related perks and abilities can now be reliably removed via the antagonist datum.
Additionally, fixes silently added and pseudo antagonist roles announcing when they are removed, alongside some other minor bugfixes and cleanup.

## Why's this needed?
Same rationale as #9366.

### CODE REVIEW

**Review by flappybatpal — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/__antagonist.dm [hunk 1]
```diff
@@ -57,13 +59,13 @@ ABSTRACT_TYPE(/datum/antagonist)
 		..()
 
 	/// Calls removal procs to soft-remove this antagonist from its owner. Actual movement or deletion of the datum still needs to happen elsewhere.
-	proc/remove_self(take_gear = TRUE, silent)
+	proc/remove_self(take_gear = TRUE)
```
[the remark this answers is not in the document]
Comment (Mister-Moriarty) [reply code/modules/antagonists/__antagonist.dm:62]: Ah, my apologies, I forgot to remove the silent argument from those calls.

**Comment by flappybatpal** [comment]
image
If a regular wrestler gets a wrestling belt they get twice the wrestle abilities. This doesn't happen on live.

**Comment by flappybatpal** [comment]
There are a further two things that might be out of scope but do need to be addressed.
1. The wrestling belt and wrestle contract both display an antagonist role on the Player Options panel:
image
While this does not show up anywhere else such as the antag list or logs it is a concern for administrators being misled if they open a Player Options panel. I imagine this is the same issue that occurs with VR antagonists, where the "false" antag role is. This might be out of scope as the lack of issue with this elsewhere points to the logic used on Player Options
2. The Antagonists panel (in Game Options) lists a Ghost Respawn wrestler twice:
image

**Review by flappybatpal — APPROVED** [review]

**Comment by TobleroneSwordfish** [comment]
I've fixed the player panel on master, still a little jank with interactions between the belt and the true antag, but it's probably fine.

**Pull request merged**